src/ray/common/buffer.h

```diff
@@ -55,9 +55,9 @@ class LocalMemoryBuffer : public Buffer {
       : has_data_copy_(copy_data) {
     if (copy_data) {
       RAY_CHECK(data != nullptr);
-      buffer_.insert(buffer_.end(), data, data + size);
-      data_ = buffer_.data();
-      size_ = buffer_.size();
+      data_ = (uint8_t *)malloc(size * sizeof(uint8_t));
```

Comment:
`malloc` and `free` are kinds of C style, could we use a `unique_ptr` instead which can auto-delete the buffer when it destructs?

Reply:
> `malloc` and `free` are kinds of C style, could we use a `unique_ptr` instead which can auto-delete the buffer when it destructs?

good idea :)